test/tests/imageserver/test_image_service_system.py

```diff
@@ -0,0 +1,319 @@
+        bmcip = response['json']['data']['IP Address']
+        if bmcip == "0.0.0.0":
+            response = fit_common.rackhdapi(
+                '/api/2.0/nodes/' + nodeid + '/catalogs/rmm')
+            bmcip = response['json']['data']['IP Address']
+        # Try credential record in config file
+        for creds in fit_common.fitcreds()['bmc']:
+            if fit_common.remote_shell(
+                'ipmitool -I lanplus -H ' + bmcip + ' -U ' +
+                    creds['username'] + ' -P ' + creds['password'] + ' fru')['exitcode'] == 0:
+                usr = creds['username']
+                pwd = creds['password']
+                break
+        # Put the credential to OBM settings
+        if usr != "":
+            payload = {
+                "service": "ipmi-obm-service",
+                "config": {
+                    "host": bmcip,
+                    "user": usr,
+                    "password": pwd},
+                "nodeId": nodeid}
+            api_data = fit_common.rackhdapi("/api/2.0/obms", action='put', payload=payload)
+            if api_data['status'] == 201:
+                return True
+        return False
+
+    def _upload_os_by_network(self, osname, osversion, source_url):
+        mon_url = '/images?name=' + osname + '&version=' + osversion + '&isoweb=' + source_url
+        serverip = self._get_serverip()
+        response = fit_common.restful(
+            "http://" +
+            serverip +
+            ":7070" +
+            mon_url,
+            rest_action="put",
+            rest_payload={},
+            rest_timeout=None,
+            rest_headers={})
+        if response['status'] in range(200, 205):
+            return response['json']
+        else:
+            logs.error('Incorrect HTTP return code, expected 201, got:' + str(response['status']))
+            return "fail"
+
+    def _list_os_image(self):
+        mon_url = '/images'
+        serverip = self._get_serverip()
+        response = fit_common.restful("http://" + serverip + ":7070" + mon_url)
+        if response['status'] in range(200, 205):
+            return response['json']
+        else:
+            logs.error('Incorrect HTTP return code, expected 201, got:' + str(response['status']))
+            return "fail"
+
+    def _list_os_iso(self):
+        mon_url = '/iso'
+        serverip = self._get_serverip()
+        response = fit_common.restful("http://" + serverip + ":7070" + mon_url)
+        if response['status'] in range(200, 205):
+            return response['json']
+        else:
+            logs.error('Incorrect HTTP return code, expected 201, got:' + str(response['status']))
+            return "fail"
+
+    def _delete_os_image(self, osname, osversion):
+        mon_url = '/images?name=' + osname + '&version=' + osversion
+        serverip = self._get_serverip()
+        response = fit_common.restful("http://" + serverip + ":7070" + mon_url, rest_action="delete")

```

Comment:
why port 7070 here?  is that specific to nodes?   and it shouldn't be hard coded.
it should be added to the config file you are reading in as what port to use.  (or ports for rackhd are defined in install_config.json.